tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,214 @@
+    private static OptionSpec<String> applicationIdOption;
+    private static OptionSpec<String> sourceTopicsOption;
+    private static OptionSpec<String> intermediateTopicsOption;
+
+    private OptionSet options = null;
+
+    public int run(final String[] args) {
+        ZkUtils zkUtils = null;
+
+        int exitCode = EXIT_CODE_SUCCESS;
+        try {
+            parseArguments(args);
+
+            zkUtils = ZkUtils.apply(this.options.valueOf(zookeeperOption),
+                30000,
+                30000,
+                JaasUtils.isZkSecurityEnabled());
+
+            final List<String> allTopics = scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics());
+
+            resetSourceTopicOffsets();
+            seekToEndIntermediateTopics(zkUtils, allTopics);
+            deleteInternalTopics(zkUtils, allTopics);
+        } catch (final Exception e) {
+            exitCode = EXIT_CODE_ERROR;
+            System.err.println(e.getMessage());
+        } finally {
+            if (zkUtils != null) {
+                zkUtils.close();
+            }
+        }
+
+        return exitCode;
+    }
+
+    private void parseArguments(final String[] args) throws IOException {
+        final OptionParser optionParser = new OptionParser();
+        applicationIdOption = optionParser.accepts("application-id", "The Kafka Streams application ID (application.id)")
+            .withRequiredArg()
+            .ofType(String.class)
+            .describedAs("id")
+            .required();
+        bootstrapServerOption = optionParser.accepts("bootstrap-server", "Format: <host:port>")
+            .withRequiredArg()
+            .ofType(String.class)
+            .defaultsTo("localhost:9092")
+            .describedAs("url");
+        zookeeperOption = optionParser.accepts("zookeeper", "Format: <host:port>")
+            .withRequiredArg()
+            .ofType(String.class)
+            .defaultsTo("localhost:2181")
+            .describedAs("url");
+        sourceTopicsOption = optionParser.accepts("source-topics", "Comma separated list of user source topics")
+            .withRequiredArg()
+            .ofType(String.class)
+            .withValuesSeparatedBy(',')
+            .describedAs("list");
+        intermediateTopicsOption = optionParser.accepts("intermediate-topics", "Comma separated list of intermediate user topics")
+            .withRequiredArg()
+            .ofType(String.class)
+            .withValuesSeparatedBy(',')
+            .describedAs("list");
+
+        try {
+            this.options = optionParser.parse(args);
+        } catch (final OptionException e) {
+            optionParser.printHelpOn(System.err);
+            throw e;
+        }
+    }
+
+    private void resetSourceTopicOffsets() {
+        final List<String> sourceTopics = this.options.valuesOf(sourceTopicsOption);
+
+        if (sourceTopics.size() == 0) {
+            System.out.println("No source topics specified.");
```

Comment:
I'd clarify this log message to sth like:

> "No source topics specified, skipping resetting source topic offsets."

Otherwise the message is imho a bit confusing because it lacks context (e.g. "Is it good or bad that I didn't specify any source topics specified?")